app/models/search/taggable_query.rb

```diff
@@ -0,0 +1,39 @@
+# Shared methods for work and bookmark queries
+module TaggableQuery
+
+  def filter_ids
+    return @filter_ids if @filter_ids.present?
+    @filter_ids = options[:filter_ids] || []
+    %w(fandom rating warning category character relationship freeform).each do |tag_type|
+      if options["#{tag_type}_ids".to_sym].present?
+        ids = options["#{tag_type}_ids".to_sym]
+        @filter_ids += ids.is_a?(Array) ? ids : [ids]
+      end
+    end
+    @filter_ids += named_tags
+    @filter_ids.uniq
+  end
+
+  def exclusion_ids
+    return @exclusion_ids if @exclusion_ids.present?
+    return if options[:excluded_tag_names].blank? && options[:excluded_tag_ids].blank?
+
+    ids = options[:excluded_tag_ids] || []
+    names = options[:excluded_tag_names]&.split(",")
```

Comment:
I'd be tempted to add a .map(&:squish) here (and in named_tags) just in case someone has javascript disabled and is manually typing in tag names. But again, that might be outside the scope of the pull request, particularly since that's not the way other_tag_names is handled on the live website.